### PR TITLE
samples/tutorial: refresh tutorial README for recent changes, and add Azure

### PR DESCRIPTION
**Pull request by vangent**
It has code snippets that were never updated for things like the import path change, etc.

### CODE REVIEW

File: samples/tutorial/README.md [hunk 1]
```diff
@@ -60,20 +63,20 @@ both clouds. This will be the only step that requires cloud-specific APIs.
 ## GCP authentication & connection
 
 As a first pass, let's write the code to connect to a GCS bucket. Then, we will
-write the code to connect to an S3 bucket. In both cases, we are going to create
-a pointer to a `blob.Bucket`, the cloud-agnostic blob storage type.
+write the code to connect to S3 and Azure buckets. In both cases, we are going
```
Comment (jba) [inline samples/tutorial/README.md]: s/both/all/?

File: samples/tutorial/README.md [hunk 2]
```diff
@@ -163,16 +173,29 @@ func setupAWS(ctx context.Context, bucket string) (*blob.Bucket, error) {
         Credentials: credentials.NewEnvCredentials(),
     }
     s := session.Must(session.NewSession(c))
-    return s3blob.OpenBucket(ctx, s, bucket)
+    return s3blob.OpenBucket(ctx, s, bucket, nil)
+}
+
+// setupAzure creates a connection to Azure Storage Account using shared key
+// authorization. It assumes environment variables AZURE_STORAGE_ACCOUNT_NAME
+// and AZURE_STORAGE_ACCOUNT_NAME are present.
```
Comment (jba) [inline samples/tutorial/README.md]: s/NAME/KEY/

File: samples/tutorial/setup.go [hunk 3]
```diff
@@ -74,14 +74,15 @@ func setupAWS(ctx context.Context, bucket string) (*blob.Bucket, error) {
 	return s3blob.OpenBucket(ctx, s, bucket, nil)
 }
 
-// setupAzure creates a connection to Azure Storage Account using shared key authorization.
-// This operation assumes environment variables AZURE_STORAGE_ACCOUNT_NAME and AZURE_STORAGE_ACCOUNT_NAME are present.
+// setupAzure creates a connection to Azure Storage Account using shared key
+// authorization. It assumes environment variables AZURE_STORAGE_ACCOUNT_NAME
+// and AZURE_STORAGE_ACCOUNT_NAME are present.
```
Comment (jba) [inline samples/tutorial/setup.go]: ditto

File: samples/tutorial/README.md (hunk 1 above)
Comment (vangent) [reply samples/tutorial/README.md]: Done.

File: samples/tutorial/README.md (hunk 2 above)
Comment (vangent) [reply samples/tutorial/README.md]: Done.

File: samples/tutorial/setup.go (hunk 3 above)
Comment (vangent) [reply samples/tutorial/setup.go]: Done.

**Comment by vangent** [comment]
PTAL.

**Pull request merged**